k2/python/k2/fsa_algo.py

```diff
@@ -252,9 +251,13 @@ def compose(a_fsa: Fsa,
     if inner_labels is not None:
         # out_fsa.`inner_labels` = out_fsa.labels
         setattr(out_fsa, inner_labels, out_fsa.labels)
-    out_fsa.aux_labels = (index(b_fsa.aux_labels, b_arc_map)
-                          if hasattr(b_fsa, 'aux_labels')
-                          else out_fsa.labels)
+
+    if hasattr(b_fsa, 'aux_labels'):
+        out_fsa.aux_labels = index(b_fsa.aux_labels, b_arc_map)
+    else:
+        # need a clone here since `Fsa.labels` is a reference
+        out_fsa.aux_labels = out_fsa.labels.clone()
+
     out_fsa.labels = index(a_fsa_inv.aux_labels, a_arc_map)
```

Comment:
Without `clone()`, `out_fsa.aux_labels` shares the same memory with `out_fsa.labels`. Any change done to either one will be reflected in the other one. `out_fsa` is always an acceptor.